azurelinuxagent/common/event.py

```diff
@@ -393,52 +529,106 @@ def _clean_up_message(message):
             else:
                 return message
 
+    def _add_common_event_parameters(self, event, event_timestamp):
+        """
+        This method is called for all events and ensures all telemetry fields are added before the event is sent out.
+        Note that the event timestamp is saved in the OpcodeName field.
+        """
+        common_params = [TelemetryEventParam('GAVersion', CURRENT_AGENT),
+                         TelemetryEventParam('ContainerId', GoalState.ContainerID),
+                         TelemetryEventParam('OpcodeName', event_timestamp.strftime(u'%Y-%m-%dT%H:%M:%S.%fZ')),
+                         TelemetryEventParam('EventTid', threading.current_thread().ident),
+                         TelemetryEventParam('EventPid', os.getpid()),
+                         TelemetryEventParam('TaskName', threading.current_thread().getName()),
+                         TelemetryEventParam('KeywordName', ''),
+                         TelemetryEventParam('ExtensionType', event.file_type),
+                         TelemetryEventParam('IsInternal', False)]
+
+        event.parameters.extend(common_params)
+        event.parameters.extend(self._common_parameters)
+
     @staticmethod
-    def add_default_parameters_to_event(event_parameters, set_values_for_agent=True):
+    def _trim_extension_event_parameters(event):
```

Comment:
This comes from #1757; we need to improve the way we handle parameters to avoid all those sequential traversals of the array